processing/src/main/java/io/druid/query/topn/TopNQueryQueryToolChest.java

```diff
@@ -284,8 +281,7 @@ public String apply(@Nullable AggregatorFactory input)
                       values.put(name, fn.manipulate(aggregatorFactories[i], input.getMetric(name)));
                     }
 
-                    final Object dimValue = input.getDimensionValue(dimension);
-                    values.put(dimension, extractionFn == null ? dimValue : extractionFn.apply(dimValue));
+                    values.put(dimension, input.getDimensionValue(dimension));
```

Comment:
No, not here. it is either being applied at a deeper (column select) or higher (post merge manipulator) level than here.